easybuild/easyblocks/t/trilinos.py

```diff
@@ -98,11 +98,8 @@ def configure_step(self):
         else:
             self.cfg.update('configopts', "-DBUILD_SHARED_LIBS:BOOL=OFF")
 
-        # release or debug get_version
-        if self.toolchain.options['debug']:
-            self.cfg.update('configopts', "-DCMAKE_BUILD_TYPE:STRING=DEBUG")
-        else:
-            self.cfg.update('configopts', "-DCMAKE_BUILD_TYPE:STRING=RELEASE")
+        # release or debug gversion
+        self.cfg['build_type'] = 'Debug' if self.toolchain.options.get('debug', None) else 'Release'
```

Comment:
@boegel I think it makes sense to do this by default. Is the toolchainopt debug used for anything else?

Reply:
We may use it in some places to inject `-g` into `$CXXFLAGS` & co for example.

Always switching to `-DDCMAKE_BUILD_TYPE=Debug` when the `debug` toolchain option is enabled makes sense I guess...

Reply:
The question would be: How to handle 'debug' vs 'build_type'? My proposal: if build_type is not None, it overwrites 'debug'. Else it is used as it is here: Debug or Release
But this borders into the discussion in https://github.com/easybuilders/easybuild-easyblocks/pull/911#issuecomment-564466693: What for projects failing to work in "Release" mode? TBH I think this is a non-issue as optflags are usually used so only difference is (almost always) the added `-DNDEBUG`

To be conservative one could treat debug=None as not doing anything which makes it:
- if build_type is not None, use it
- else if debug is not None, use Release or Debug
- don't use anything

But then: Which ECs use `debug: False` explicitly? So I'd treat debug=None as false and if anything breaks, set it in the EC (again because I see very small chances for anything in ECs break)

Reply:
On the other hand current behavior for e.g. gromacs and netcdf is that toolchainopts has preference over build_type. Maybe use that?